### PR TITLE
host-ctr: upgrades dependencies for v1.13.0

### PR DESCRIPTION
**Pull request by jpmcb**
**Issue number:**

Related to https://github.com/bottlerocket-os/bottlerocket/issues/2761

**Description of changes:**

```
host-ctr: upgrades Go dependencies

- Upgrades containerd code dependencies to v1.6.19
- Upgrades opencontainers/runtime-spec to one used by containerd v1.6.19
- Upgrades golang.org/x/net and golang.org/x/crypto dependencies
  (which are not directly used) out of band of them being updated in
  aws-sdk-go. See the following for more info:
  https://github.com/aws/aws-sdk-go/issues/4738
```

**Testing done:**

_Testing incoming!_

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Comment by jpmcb** [comment]
Host containers look 👍🏼 

```
bash-5.1# systemctl status host-containers@admin
● host-containers@admin.service - Host container: admin
     Loaded: loaded (/x86_64-bottlerocket-linux-gnu/sys-root/usr/lib/systemd/system/host-containers@.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2023-02-23 22:51:50 UTC; 3min 43s ago
   Main PID: 2199 (host-ctr)
      Tasks: 9 (limit: 9162)
     Memory: 16.2M
     CGroup: /system.slice/system-host\x2dcontainers.slice/host-containers@admin.service
             └─ 2199 /usr/bin/host-ctr run --container-id=admin --source=328549459982.dkr.ecr.us-west-2.amazonaws.com/bottlerocket-admin:v0.9.4 --superpowered=true --registry-config=/etc/host-containers/host-ctr.toml

Feb 23 22:51:51 ip-172-31-56-110.us-west-2.compute.internal host-ctr[2199]: time="2023-02-23T22:51:51Z" level=info msg="pulled image successfully" img="ecr.aws/arn:aws:ecr:us-west-2:328549459982:repository/bottlerocket-admin:v0.9.4"
Feb 23 22:51:51 ip-172-31-56-110.us-west-2.compute.internal host-ctr[2199]: time="2023-02-23T22:51:51Z" level=info msg="unpacking image..." img="ecr.aws/arn:aws:ecr:us-west-2:328549459982:repository/bottlerocket-admin:v0.9.4"
Feb 23 22:51:53 ip-172-31-56-110.us-west-2.compute.internal host-ctr[2199]: time="2023-02-23T22:51:53Z" level=info msg="tagging image" img="328549459982.dkr.ecr.us-west-2.amazonaws.com/bottlerocket-admin:v0.9.4"
Feb 23 22:51:53 ip-172-31-56-110.us-west-2.compute.internal host-ctr[2199]: time="2023-02-23T22:51:53Z" level=info msg="Container does not exist, proceeding to create it" ctr-id=admin
Feb 23 22:51:53 ip-172-31-56-110.us-west-2.compute.internal host-ctr[2199]: time="2023-02-23T22:51:53Z" level=info msg="container task does not exist, proceeding to create it" container-id=admin
Feb 23 22:51:53 ip-172-31-56-110.us-west-2.compute.internal host-ctr[2199]: time="2023-02-23T22:51:53Z" level=info msg="successfully started container task"
Feb 23 22:51:53 ip-172-31-56-110.us-west-2.compute.internal host-ctr[2199]: Created symlink /root/.config/systemd/user/admin.target.wants/getty@tty0.service, pointing to /etc/systemd/user/getty@.service.
Feb 23 22:51:53 ip-172-31-56-110.us-west-2.compute.internal host-ctr[2199]: Created symlink /root/.config/systemd/user/admin.target.wants/serial-getty@ttyS0.service, pointing to /etc/systemd/user/serial-getty@.service.
Feb 23 22:51:53 ip-172-31-56-110.us-west-2.compute.internal host-ctr[2199]: Created symlink /root/.config/systemd/user/admin.target.wants/sshd.service, pointing to /etc/systemd/user/sshd.service.
Feb 23 22:51:53 ip-172-31-56-110.us-west-2.compute.internal host-ctr[2199]: Startup finished in 20ms.

bash-5.1# systemctl status host-containers@control
● host-containers@control.service - Host container: control
     Loaded: loaded (/x86_64-bottlerocket-linux-gnu/sys-root/usr/lib/systemd/system/host-containers@.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2023-02-23 22:50:18 UTC; 5min ago
   Main PID: 1749 (host-ctr)
      Tasks: 9 (limit: 9162)
     Memory: 55.7M
     CGroup: /system.slice/system-host\x2dcontainers.slice/host-containers@control.service
             └─ 1749 /usr/bin/host-ctr run --container-id=control --source=328549459982.dkr.ecr.us-west-2.amazonaws.com/bottlerocket-control:v0.7.0 --superpowered=false --registry-config=/etc/host-containers/host-ctr.toml

Feb 23 22:50:24 ip-172-31-56-110.us-west-2.compute.internal host-ctr[1749]: 2023-02-23 22:50:24 INFO [amazon-ssm-agent] registrar detected. Attempting registration
Feb 23 22:50:24 ip-172-31-56-110.us-west-2.compute.internal host-ctr[1749]: 2023-02-23 22:50:24 INFO [Registrar] Starting registrar module
Feb 23 22:50:24 ip-172-31-56-110.us-west-2.compute.internal host-ctr[1749]: 2023-02-23 22:50:24 INFO [EC2Identity] no registration info found for ec2 instance, attempting registration
Feb 23 22:50:24 ip-172-31-56-110.us-west-2.compute.internal host-ctr[1749]: 2023-02-23 22:50:24 INFO [CredentialRefresher] credentialRefresher has started
Feb 23 22:50:24 ip-172-31-56-110.us-west-2.compute.internal host-ctr[1749]: 2023-02-23 22:50:24 INFO [CredentialRefresher] Starting credentials refresher loop
Feb 23 22:50:24 ip-172-31-56-110.us-west-2.compute.internal host-ctr[1749]: 2023-02-23 22:50:24 INFO EC2RoleProvider Successfully connected with instance profile role credentials
Feb 23 22:50:24 ip-172-31-56-110.us-west-2.compute.internal host-ctr[1749]: 2023-02-23 22:50:24 INFO [CredentialRefresher] Next credential rotation will be in 30.599988057866668 minutes
Feb 23 22:50:25 ip-172-31-56-110.us-west-2.compute.internal host-ctr[1749]: 2023-02-23 22:50:25 INFO [amazon-ssm-agent] [LongRunningWorkerContainer] [WorkerProvider] Worker ssm-agent-worker is not running, starting worker process
Feb 23 22:50:25 ip-172-31-56-110.us-west-2.compute.internal host-ctr[1749]: 2023-02-23 22:50:25 INFO [amazon-ssm-agent] [LongRunningWorkerContainer] [WorkerProvider] Worker ssm-agent-worker (pid:19) started
Feb 23 22:50:25 ip-172-31-56-110.us-west-2.compute.internal host-ctr[1749]: 2023-02-23 22:50:25 INFO [amazon-ssm-agent] [LongRunningWorkerContainer] Monitor long running worker health every 60 seconds
```

**Review by stmcginnis — APPROVED** [review]

**Review by webern — APPROVED** [review]

**Comment by jpmcb** [comment]
Force pushed to fix grammar in commit message ✍🏼

File: sources/host-ctr/go.mod [hunk 1]
```diff
@@ -3,16 +3,16 @@ module host-ctr
 go 1.19
 
 require (
-	github.com/aws/aws-sdk-go v1.44.176
+	github.com/aws/aws-sdk-go v1.44.208
 	github.com/awslabs/amazon-ecr-containerd-resolver v0.0.0-20221221205310-1903c4ed45d1
 	github.com/containerd/containerd v1.6.15
```
Comment (etungsten) [inline sources/host-ctr/go.mod]: I think we should rebase this PR after https://github.com/bottlerocket-os/bottlerocket/pull/2825 gets merged so we can move to containerd 1.6.18 along with the other dep updates in one go.

Comment (jpmcb) [reply sources/host-ctr/go.mod]: That's a good call - yes; if we're upgrading to containerd 1.6.18, we can wait for that to get in and rebase on that to upgrade those dependencies too

**Comment by jpmcb** [comment]
Force pushed to:
- add out of band `/x/net` and `/x/crypto` upgrades
- upgrade `continerd` to `v1.6.18` (the version being upgraded to in #2825)
- upgrade open contianers runtime spec to version used by newer contaienrd

**Comment by jpmcb** [comment]
Force pushed to upgrade to using the newly cut containerd `v1.6.19`

**Review by etungsten — APPROVED** [review]

**Comment by jpmcb** [comment]
Re-built and re-ran tests. Everything's looking good. Will merge today unless someone objects:
```
bash-5.1# systemctl status host-containers@admin
● host-containers@admin.service - Host container: admin
     Loaded: loaded (/x86_64-bottlerocket-linux-gnu/sys-root/usr/lib/systemd/system/host-containers@.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2023-03-02 17:26:29 UTC; 20s ago
   Main PID: 2242 (host-ctr)
      Tasks: 10 (limit: 9162)
     Memory: 19.9M
     CGroup: /system.slice/system-host\x2dcontainers.slice/host-containers@admin.service
             └─ 2242 /usr/bin/host-ctr run --container-id=admin --source=328549459982.dkr.ecr.us-west-2.amazonaws.com/bottlerocket-admin:v0.9.4 --superpowered=true --registry-config=/etc/host-containers/host-ctr.toml

Mar 02 17:26:30 ip-172-31-58-189.us-west-2.compute.internal host-ctr[2242]: time="2023-03-02T17:26:30Z" level=info msg="pulled image successfully" img="ecr.aws/arn:aws:ecr:us-west-2:328549459982:repository/bottlerocket-admin:v0.9.4"
Mar 02 17:26:30 ip-172-31-58-189.us-west-2.compute.internal host-ctr[2242]: time="2023-03-02T17:26:30Z" level=info msg="unpacking image..." img="ecr.aws/arn:aws:ecr:us-west-2:328549459982:repository/bottlerocket-admin:v0.9.4"
Mar 02 17:26:32 ip-172-31-58-189.us-west-2.compute.internal host-ctr[2242]: time="2023-03-02T17:26:32Z" level=info msg="tagging image" img="328549459982.dkr.ecr.us-west-2.amazonaws.com/bottlerocket-admin:v0.9.4"
Mar 02 17:26:32 ip-172-31-58-189.us-west-2.compute.internal host-ctr[2242]: time="2023-03-02T17:26:32Z" level=info msg="Container does not exist, proceeding to create it" ctr-id=admin
Mar 02 17:26:32 ip-172-31-58-189.us-west-2.compute.internal host-ctr[2242]: time="2023-03-02T17:26:32Z" level=info msg="container task does not exist, proceeding to create it" container-id=admin
Mar 02 17:26:32 ip-172-31-58-189.us-west-2.compute.internal host-ctr[2242]: time="2023-03-02T17:26:32Z" level=info msg="successfully started container task"
Mar 02 17:26:32 ip-172-31-58-189.us-west-2.compute.internal host-ctr[2242]: Created symlink /root/.config/systemd/user/admin.target.wants/getty@tty0.service, pointing to /etc/systemd/user/getty@.service.
Mar 02 17:26:32 ip-172-31-58-189.us-west-2.compute.internal host-ctr[2242]: Created symlink /root/.config/systemd/user/admin.target.wants/serial-getty@ttyS0.service, pointing to /etc/systemd/user/serial-getty@.service.
Mar 02 17:26:32 ip-172-31-58-189.us-west-2.compute.internal host-ctr[2242]: Created symlink /root/.config/systemd/user/admin.target.wants/sshd.service, pointing to /etc/systemd/user/sshd.service.
Mar 02 17:26:32 ip-172-31-58-189.us-west-2.compute.internal host-ctr[2242]: Startup finished in 19ms.
bash-5.1# systemctl status host-containers@control
● host-containers@control.service - Host container: control
     Loaded: loaded (/x86_64-bottlerocket-linux-gnu/sys-root/usr/lib/systemd/system/host-containers@.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2023-03-02 16:49:10 UTC; 37min ago
   Main PID: 1745 (host-ctr)
      Tasks: 10 (limit: 9162)
     Memory: 55.2M
     CGroup: /system.slice/system-host\x2dcontainers.slice/host-containers@control.service
             └─ 1745 /usr/bin/host-ctr run --container-id=control --source=328549459982.dkr.ecr.us-west-2.amazonaws.com/bottlerocket-control:v0.7.0 --superpowered=false --registry-config=/etc/host-containers/host-ctr.toml

Mar 02 16:49:17 ip-172-31-58-189.us-west-2.compute.internal host-ctr[1745]: 2023-03-02 16:49:15 INFO [EC2Identity] no registration info found for ec2 instance, attempting registration
Mar 02 16:49:31 ip-172-31-58-189.us-west-2.compute.internal host-ctr[1745]: 2023-03-02 16:49:31 INFO [CredentialRefresher] credentialRefresher has started
Mar 02 16:49:31 ip-172-31-58-189.us-west-2.compute.internal host-ctr[1745]: 2023-03-02 16:49:31 INFO [CredentialRefresher] Starting credentials refresher loop
Mar 02 16:49:31 ip-172-31-58-189.us-west-2.compute.internal host-ctr[1745]: 2023-03-02 16:49:31 INFO EC2RoleProvider Successfully connected with instance profile role credentials
Mar 02 16:49:31 ip-172-31-58-189.us-west-2.compute.internal host-ctr[1745]: 2023-03-02 16:49:31 INFO [CredentialRefresher] Next credential rotation will be in 31.8583300854 minutes
Mar 02 16:49:32 ip-172-31-58-189.us-west-2.compute.internal host-ctr[1745]: 2023-03-02 16:49:32 INFO [amazon-ssm-agent] [LongRunningWorkerContainer] [WorkerProvider] Worker ssm-agent-worker is not running, starting worker process
Mar 02 16:49:32 ip-172-31-58-189.us-west-2.compute.internal host-ctr[1745]: 2023-03-02 16:49:32 INFO [amazon-ssm-agent] [LongRunningWorkerContainer] [WorkerProvider] Worker ssm-agent-worker (pid:17) started
Mar 02 16:49:32 ip-172-31-58-189.us-west-2.compute.internal host-ctr[1745]: 2023-03-02 16:49:32 INFO [amazon-ssm-agent] [LongRunningWorkerContainer] Monitor long running worker health every 60 seconds
Mar 02 17:21:23 ip-172-31-58-189.us-west-2.compute.internal host-ctr[1745]: 2023-03-02 17:21:23 INFO EC2RoleProvider Successfully connected with instance profile role credentials
Mar 02 17:21:23 ip-172-31-58-189.us-west-2.compute.internal host-ctr[1745]: 2023-03-02 17:21:23 INFO [CredentialRefresher] Next credential rotation will be in 31.3083306597 minutes
bash-5.1#
```

**Pull request merged**